src/server/identity/server/api_server.go

```diff
@@ -37,10 +38,61 @@ func NewIdentityServer(env Env, public bool) identity.APIServer {
 		}()
 	}
 
-	return server
+	return server, server.envBootstrap
+}
+
+func (a *apiServer) envBootstrap(ctx context.Context) error {
+	if a.env.Config.IdentityConfig != "" {
+		a.env.Logger.Info("Started to configure identity server config via environment")
+		var config identity.IdentityServerConfig
+		if err := yaml.Unmarshal([]byte(a.env.Config.IdentityConfig), &config); err != nil {
+			return errors.Wrapf(err, "failed to unmarshal IdentityConfig with value: %s.", a.env.Config.IdentityConfig)
```

Comment:
minor nit; I'd avoid "failed" in error objects.  would probably say `errors.Wrapf(err, "unmarshal IdentityServerConfig %q", a.env.Config.IdentityConfig)`.